### DISCUSSION TITLE
About Document dates being inaccurate

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#6924] [Q&A] [answered]
My deployment platform is Vercel (primary), and two secondary ones: Netlify, cloudflare pages
My deployment variables on these three platforms are CLONE_DEPTH=full
I realized that after I set
```
---
date:
   created: 2024-02-29
   updated: 2024-03-17
---
```
But the time displayed on the actual test page (local) is not accurate.

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/0d788662-cacc-42e3-a367-ef11b58e472c)

There is another question. Is there a way to achieve this? When I do not specify the date in the Markdown metadata, the plug-in automatically obtains the creation time and modification time of my md file and displays it on the page.

https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/issues/118
I found a similar problem, but his local is normal, my local and deployment environments are not normal, and I can't find whether the official document mentions the CI environment parameters of the three deployment platforms I use.

**Comment by w1ndys** [comment]
I also found that in Netlify, even if no environment variables are specified, it will be displayed normally, but not in Vercel and cloudflare. So far, I have not found variables that adapt to these two abnormal servers.

**Reply by w1ndys** [reply]
I've been searching for three hours. My head is going to explode. I really can't find a suitable solution😭😭😭😭

**Comment by kamilkrzyskow** [comment]
Hello @W1ndys,
the date syntax:
```yml
---
date:
   created: 2024-02-29
   updated: 2024-03-17
---
```
is only applicable for the blog plugin and is displayed on the left hand side of a blog post:
https://squidfunk.github.io/mkdocs-material/blog/2023/11/30/adding-a-badge-to-your-project/
(see the metadata on the left)

The screenshot you provided is the time data at the bottom of the page provided by the git-revision plugin. 
Typically servers use UTC timezone, and I'm assuming you're basing your time from China, so that would explain your one day difference.

The plugin has an option to set a timezone, so try that first:
https://timvink.github.io/mkdocs-git-revision-date-localized-plugin/options/#timezone

I'm unfamiliar with Vercel or Cloudflare CI, but there surely is an option to change the timezone there as well, if that plugin option doesn't work.

**Reply by w1ndys** [reply]
First of all, thank you very much for your reply. I tried setting the time zone to Asia/Shanghai, but I found that it had no effect, so I guess this is related to the environment variables of deploying CI, because the results displayed when I deployed on Netlify were different from those deployed on Vercel.  , I guess Netlify is correct, but Vercel needs some kind of environment variable to display correctly, but I can't find a way around it.  I guess I should reiterate that when I mention the metadata date, I'm not referring to the metadata on the left side of the blog (next to the avatar), I'm referring to the one at the bottom of the documentation page (like the plugin page), next to the contributor who created it  date and modification date.  I would also like to ask where the display of these two dates is based?  Metadata or other basis.  If it is metadata, is there a way to determine the display result just by the creation and modification time of the md source file (instead of having to write metadata, which I think is a waste if it has to be written for every file)  Time thing) I also found a situation in the results of my debugging: I did not write metadata about the date, but the results showed the time, which was different from my file creation time and modification time.  If you want more information I'll be happy to provide it to you

**Reply by kamilkrzyskow** [reply]
> I'm referring to the one at the bottom of the documentation page (like the plugin page), next to the contributor who created it date and modification date. I would also like to ask where the display of these two dates is based?

The time at the bottom of the page is based from the information provided by the plugin.

https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/af27290b2d88a309903b786e60c0aba0e7cae674/mkdocs_git_revision_date_localized_plugin/plugin.py#L220-L222

https://github.com/squidfunk/mkdocs-material/blob/022f14c0ebc757eae6eefacb5b5ccb140a443e0f/src/templates/partials/source-file.html#L134-L146

> I tried setting the time zone to Asia/Shanghai, but I found that it had no effect

I never used Vercel, but perhaps it caches previous CI runs, and the update didn't have any effect. You could also run a Python hook that runs systems commands to check the timezone and maybe modify it 🤔 https://www.mkdocs.org/user-guide/configuration/#hooks

The revision plugin could also be debugged in the CI via a hook.

---

You could also build the site with GitHub Actions CI and then deploy the already built files into Vercel, so the timezone etc. would be handled with the GitHub Actions.

**Reply by w1ndys** [reply]
thank you for your information. I would try using GitHub actions CI and transition the generated page directly to Vercel
In addition, I think the key point is to find the matching environment variables on Vercel or cloudflare pages. The variables for the three officially mentioned platforms are as follows
https://github.com/timvink/mkdocs-git-revision-date-localized-plugin?tab=readme-ov-file#note-when-using-build-environments

![Screenshot_2024-03-18-08-02-42-023_com android chrome](https://github.com/squidfunk/mkdocs-material/assets/109416673/05c39783-8dec-4878-82a1-80b566f42166)


If it still doesn't work, I choose to give up solving it, because the problem is very subtle and has little impact on the system, but it consumes too much energy on our part.

![Screenshot_2024-03-18-08-03-17-959_com android chrome](https://github.com/squidfunk/mkdocs-material/assets/109416673/cc6d018f-f451-4589-a144-b122990e8da1)

I also ignored some information. This is the Vercel deployment log, which mentioned relevant content, so this is also the key point I mentioned above, which is the "depth" in the log, but I don't know how to name this variable and its value (full Or 0 etc.)

**Reply by kamilkrzyskow** [reply, marked as answer]
In this discussion: 
- https://github.com/vercel/vercel/discussions/5737

It's said that Vercel automatically fetches a shallow clone of the repository. You can't change that, but it seems you can run `git` commands to unshallow the local copy or maybe fetch the full depth again:
- https://gitlab.com/Fryuni/blog/-/commit/3719380d11b2b7f752e92ac55f3a6ad4ac3d6ed8#bd6fd32b14ba32be12c1deb732d1dbaa322bb51b_0_57

Though, it's not in the scope of my expertise, so it's up to you if you want to fiddle with it more.

**Comment by w1ndys** [comment]
@kamilkrzyskow 
If we can't solve it, I think there is a wonderful way. I manually edit the Markdown source file, add the words and values of the creation date and modification date to the source file, and uninstall the plug-in, so that recording can also be achieved. Date function (although this work is a bit boring, it can ensure that whatever I write will be generated)
for example
In `test.md`
```
This is the data content
...........
created：date1
updated：date2
```
This is really a way

**Comment by w1ndys** [comment]
@kamilkrzyskow 
Good news for you, I think I've identified a way to replace the environmental problems we can't solve, as I said above

I achieved the effect I wanted by editing the icons and words at the bottom of the Markdown file (although this method is relatively inefficient and boring, I think it is much better than not solving the problem)

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/9ab9a7a5-f569-46fd-846f-118335ebfce1)
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/13ececcd-3f69-448e-8dd1-1aad7e4464f6)

And I also learned a piece of knowledge related to CSS, about how to control color attributes separately in light and dark modes (the basis for my understanding of this knowledge is through a question I asked a few days ago)，here：https://github.com/squidfunk/mkdocs-material/discussions/6882#/

![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/c7cefef6-6dae-4104-8b64-8ee5c479457b)

I'm really happy to learn this knowledge. Thank you for your material's contribution to the world.

**Reply by kamilkrzyskow** [reply]
You can't change the environmental variables, but there seems to be a solution other than that: https://github.com/squidfunk/mkdocs-material/discussions/6924#discussioncomment-8834059
Also in the linked thread about the unshallow action, there is a github action provided, which you could reference to run the build in actions and deploy the result to vercel.

I don't know how many posts you plan to write on your site. If it's many then, I recommend getting an automatic workflow to work, as changing the files manually would simply be more error-prone, aside from being tedious.

**Reply by w1ndys** [reply]
I'm sorry that I was busy yesterday and didn't reply in time. I will try the method you mentioned to solve the problem of shallow cloning.
If that doesn't work, I'll use the method I mentioned above instead. As you said, this method is not only tedious, but also error-prone. Now I have an idea to reduce this impact. I can implement "another automated workflow" through Python scripts to reduce errors caused by manual modifications. The general idea is to design two Python scripts. One is used to create and add the relevant date to the end of the content when creating; the other is used to modify and update the date at the end of the content by getting the date attribute of the md file.
If you have better ideas, I'd be happy to communicate with you

**Comment by w1ndys** [comment]
However, I still have a small problem
In the icon library I found "update" which represents the update date
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/8b99a66e-0008-4d6c-99cc-d414a9e7c31a)
But I can't find the icon representing the creation date.
![image](https://github.com/squidfunk/mkdocs-material/assets/109416673/edef656f-abde-4dc2-bd8b-13e5b525bae9) 
Can you please tell me where it is?

**Reply by w1ndys** [reply]
Now I have solved this problem by saving and customizing, I would also be grateful if you could help me find the online icon
https://squidfunk.github.io/mkdocs-material/setup/changing-the-logo-and-icons/#customization

**Reply by kamilkrzyskow** [reply]
The icons are `material-clock-edit-outline`:
https://github.com/squidfunk/mkdocs-material/blob/022f14c0ebc757eae6eefacb5b5ccb140a443e0f/src/templates/partials/source-file.html#L27
and `material-clock-plus-outline`:
https://github.com/squidfunk/mkdocs-material/blob/022f14c0ebc757eae6eefacb5b5ccb140a443e0f/src/templates/partials/source-file.html#L37
